src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -102,41 +223,49 @@ export class SavedObjectsService
       savedObjectValidations,
       logger: this.coreContext.logger.get('migrations'),
       kibanaVersion: this.coreContext.env.packageInfo.version,
-      config: coreSetup.legacy.pluginExtendedConfig,
+      config: setupDeps.legacyPlugins.pluginExtendedConfig,
       savedObjectsConfig,
       kibanaConfig,
       callCluster: retryCallCluster(adminClient.callAsInternalUser),
     }));
 
-    const mappings = this.migrator.getActiveMappings();
-    const allTypes = Object.keys(getRootPropertiesObjects(mappings));
-    const schema = new SavedObjectsSchema(savedObjectSchemas);
-    const serializer = new SavedObjectsSerializer(schema);
-    const visibleTypes = allTypes.filter(type => !schema.isHiddenType(type));
+    const createSORepository = (
+      callCluster: (
+        endpoint: string,
+        clientParams: Record<string, any>,
+        options?: CallAPIOptions
+      ) => Promise<any>,
```

Comment:
We actually have a type defined for this called `APICaller` in `src/core/server/elasticsearch/api_types`